pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -190,7 +190,7 @@ def restore_loops(self) -> None:
         self.trainer.fit_loop.current_epoch = self._loaded_checkpoint["epoch"]
 
         # crash if max_epochs is lower then the current epoch from the checkpoint
-        if self.trainer.max_epochs is not None and self.trainer.current_epoch > self.trainer.max_epochs:
+        if self.trainer.max_epochs not in (None, -1) and self.trainer.current_epoch > self.trainer.max_epochs:
```

Comment:
since `(None, -1)` is repeated in 3 places, should we have some internal constant for this?

Reply:
I made it a static method for `FitLoop`. Thoughts?